test/unit/app/messaging/fetch-backoffice-appliation.test.js

```diff
@@ -0,0 +1,54 @@
+const fetchBackOfficeApplication = require('../../../../app/messaging/fetch-backoffice-application')
+const { getBackOfficeApplicationResponseMsgType, backOfficeResponseQueue } = require('../../../../app/config')
+const states = require('../../../../app/messaging/states')
+
+jest.mock('../../../../app/lib/send-email')
+const sendMessage = require('../../../../app/messaging/send-message')
+jest.mock('../../../../app/messaging/send-message')
+const applicationRepository = require('../../../../app/repositories/application-repository')
+jest.mock('../../../../app/repositories/application-repository')
+
+describe(('Store backOffice in database'), () => {
+  const sessionId = '8e5b5789-dad5-4f16-b4dc-bf6db90ce090'
+  const email = 'email@domain.com'
+  const name = 'name-on-org'
+  const message = {
+    body: {
+      organisation: {
+        email,
+        name
+      }
+    },
+    sessionId
+  }
+
+  beforeEach(async () => {
+    jest.clearAllMocks()
+  })
+
+  const reference = '23D13'
+  test('successfully process backOffice get application request', async () => {
+    applicationRepository.get.mockResolvedValue({
+      dataValues: {
+        reference,
+        data: message.body,
+        createdBy: 'admin',
+        createdAt: new Date()
+      }
+    })
+
+    await fetchBackOfficeApplication(message)
+
+    expect(applicationRepository.get).toHaveBeenCalledTimes(1)
+    expect(sendMessage).toHaveBeenCalledTimes(1)
+    // expect(sendMessage).toHaveBeenCalledWith({ applicationState: states.submitted, backOfficeReference: reference }, getBackOfficeApplicationResponseMsgType, backOfficeResponseQueue, { sessionId })
```

Comment:
The expectation for the call to send message needs uncommenting?

Also, it would be good to check the call made to `applicationRepository.get` e.g.
```js
expect(applicationRepository.get).toHaveBeenCalledWith(reference)
```

Reply:
Please implement this expectation.

Reply:
Done